src/hlsBinaries.ts

```diff
@@ -266,21 +270,42 @@ export async function findHaskellLanguageServer(
   } else {
     // we manage HLS, make sure ghcup is installed/available
     await upgradeGHCup(context, logger);
+    
+    const toolchainConfig: ToolConfig = new Map(Object.entries(workspace.getConfiguration('haskell').get('toolchain') as any)) as ToolConfig;
+    let latestHLS:   string | undefined | null = undefined;
```

Comment:
Huh, I wasn't aware `undefined` and `null` were different values in JS.

Reply:
Yeah, I'm abusing this fact to force different codepath. Great engineering.